### DISCUSSION TITLE
Permisions for Cusctom Scripts and Reports

### DISCUSSION THREAD

**Discussion by bogdancordos** [netbox-community/netbox#9002] [Ideas]
I might not be the only one that would find useful, splitting "_extras | script_" and "_extras | report_ " into  "extras | script" and "_extras | script_admin_". Same for the reports. This will enable an admin to assign permissions to some scripts/reports and not to all of them.
This probably will involve creating a new paths _/opt/netbox/netbox/scripts_admin_ and _/opt/netbox/netbox/reports_admin_.

Thank you

**Comment by kkthxbye-code** [comment]
> splitting "extras | script" and "extras | report " into "extras | script" and "extras | script_admin".

I don't see this as a good solution. What if you need to have more than 2 permission groups?

Doing permission handling from the script is the best option right now. So something like this:

```
    def run(self, data, commit):
        if not self.request.user.groups.filter(name="can-run-script-a").exists():
            self.log_failure("Access denied")
            return

        self.log_success("Success!")
```

This will only allow users in the `can-run-script-a` group to run the script.

While I think a better option for handling script permissions is a good idea, I don't think your suggested implementation is good enough.

**Comment by bogdancordos** [comment]
> What if you need to have more than 2 permission groups?

Well, I think 2 are better than none and having them as a different model, you can actually hide those scripts from view and the source code.
I'm not saying that my suggestion is the right one. It is just that we can use some tools to manage those permissions better, as you said.
